src/query/functions/src/scalars/datetime.rs

```diff
@@ -1027,23 +1041,18 @@ fn register_to_number_functions(registry: &mut FunctionRegistry) {
 }
 
 fn register_timestamp_add_sub(registry: &mut FunctionRegistry) {
-    registry.register_2_arg::<DateType, Int64Type, DateType, _, _>(
+    registry.register_passthrough_nullable_2_arg::<DateType, Int64Type, DateType, _, _>(
         "plus",
-        |_, lhs, rhs| {
-            (|| {
-                let lm = lhs.max;
-                let ln = lhs.min;
-                let rm: i32 = num_traits::cast::cast(rhs.max)?;
-                let rn: i32 = num_traits::cast::cast(rhs.min)?;
-
-                Some(FunctionDomain::Domain(SimpleDomain::<i32> {
-                    min: ln.checked_add(rn)?,
-                    max: lm.checked_add(rm)?,
-                }))
-            })()
-            .unwrap_or(FunctionDomain::Full)
-        },
-        |a, b, _| a + (b as i32),
+        |_, _, _| FunctionDomain::MayThrow,
```

Comment:
This will affect constant folder which make example query with filter `timestamp + 3 >  t2` fail to prune partitions.